2-projects/1-todo-list/script.js

```diff
@@ -1,25 +1,84 @@
+        liEle.style.textDecoration = "line-through";
+      }
+        todo.completed =!todo.completed;
+    });
+
+    const deleteBtn = document.createElement("button");
+    deleteBtn.classList.add("badge", "bg-primary", "rounded-pill");
+    let binElement = document.createElement("i");
+    binElement.classList.add("fa", "fa-trash");
+    deleteBtn.appendChild(binElement);
+    liEle.appendChild(deleteBtn);
+
+    deleteBtn.addEventListener("click", () => {
+    listEle.removeChild(liEle);
+    });
+  }
 }
 
-// These are the same todos that currently display in the HTML
+// These are the same toDos that currently display in the HTML
 // You will want to remove the ones in the current HTML after you have created them using JavaScript
-let todos = [
+let toDos = [
   { task: "Wash the dishes", completed: false },
   { task: "Do the shopping", completed: false },
 ];
 
-populateTodoList(todos);
+populateTodoList(toDos);
+
+// This function will take the value of the input field and add it as a new todo to the bottom of the todo list. These new toDos will need the completed and delete buttons adding like normal.
+const addBtn = document.getElementsByClassName("btn btn-primary mb-3")[0];
+addBtn.addEventListener("click", addNewTodo);
 
-// This function will take the value of the input field and add it as a new todo to the bottom of the todo list. These new todos will need the completed and delete buttons adding like normal.
 function addNewTodo(event) {
   // The code below prevents the page from refreshing when we click the 'Add Todo' button.
   event.preventDefault();
-  // Write your code here... and remember to reset the input field to be blank after creating a todo!
+  // Write your code here... and remember to reset the input field to be blank aft er creating a todo!
+  let inputElement = document.getElementById("todoInput");
+  
+  if (inputElement.value === "") {
+    
+  } else {
+ let ulFlusher = document.getElementById("todo-list");
+    ulFlusher.innerHTML = "";
+    let newItem = {};
+    newItem["task"] = inputElement.value;
+    newItem["completed"] = false;
+    toDos.push(newItem);
+    populateTodoList(toDos);
+  }
 }
 
-// Advanced challenge: Write a fucntion that checks the todos in the todo list and deletes the completed ones (we can check which ones are completed by seeing if they have the line-through styling applied or not).
-function deleteAllCompletedTodos() {
+// Advanced challenge: Write a function that checks the toDos in the todo list and deletes the completed ones (we can check which ones are completed by seeing if they have the line-through styling applied or not).
+function deleteAllCompletedToDos(event) {
   // Write your code here...
+  let listEle = document.getElementById("todo-list");
+  const liEle = document.querySelectorAll("li");
+  // let child_array = Array.from(liEle.child);
+  for (let li of liEle) {
+    if (li.style.textDecoration === "line-through") {
+      listEle.removeChild(li);
+    }
```

Comment:
Instead of looping through all li elements is there something else we could loop through?